crates/dekoder/src/inventory.rs

```diff
@@ -14,7 +14,9 @@ const HEADER_EXT: &'static str = "*.yaml";
 /// Header type in an inventory.
 pub(crate) trait HeaderT {
     /// Load from yaml.
-    fn load_from_yaml(yml: &Yaml) -> Result<Box<Self>>;
+    fn load_from_yaml(yml: &Yaml) -> Result<Self>
+    where
+        Self: Sized;
```

Comment:
It should not be needed, as `Sized` is assumed by default. Instead, you should use `?Sized` if it is possible to be unsized.

But I may be missing something

Reply:
this is what the compiler suggested me and the only way it works

Reply:
Ok, it seems that a trait `Self` is the only exception to the general rule

> All type parameters have an implicit bound of Sized. The special syntax ?Sized can be used to remove this bound if it’s not appropriate.

> The one exception is the implicit Self type of a trait. A trait does not have an implicit Sized bound as this is incompatible with [trait object](https://doc.rust-lang.org/book/ch17-02-trait-objects.html)s where, by definition, the trait needs to work with all possible implementors, and thus could be any size.

https://doc.rust-lang.org/std/marker/trait.Sized.html

Reply:
With a caveat

> Although Rust will let you bind Sized to a trait, you won’t be able to use it to form a trait object later:

(same page).

But if you're not planning to use dynamic dispatch, then there is no problem :)